app/src/main/java/com/gophillygo/app/data/DestinationRepository.java

```diff
@@ -92,21 +121,71 @@ protected Void doInBackground(Void... voids) {
     }
 
     public LiveData<Resource<List<Destination>>> loadDestinations() {
-        return new AttractionNetworkBoundResource<Destination>(webservice, destinationDao, eventDao) {
+        LiveData<Resource<List<Destination>>> data = new AttractionNetworkBoundResource<Destination>(webservice, destinationDao, eventDao) {
             @NonNull
             @Override
             protected LiveData<List<Destination>> loadFromDb() {
                 return destinationDao.getAll();
             }
         }.getAsLiveData();
+        return addAttractionFlags(data, false);
     }
 
     public LiveData<Resource<List<Event>>> loadEvents() {
-        return new AttractionNetworkBoundResource<Event>(webservice, destinationDao, eventDao) {
+        LiveData<Resource<List<Event>>> data = new AttractionNetworkBoundResource<Event>(webservice, destinationDao, eventDao) {
             @NonNull @Override
             protected LiveData<List<Event>> loadFromDb() {
                 return eventDao.getAll();
             }
         }.getAsLiveData();
+        return addAttractionFlags(data, true);
+    }
+
+    private <T extends Attraction> LiveData<T> addAttractionFlag(LiveData<T> data) {
+        return switchMap(data, attraction -> {
+            MutableLiveData<T> result = new MutableLiveData<>();
+            if (attraction == null) {
+                result.postValue(attraction);
```

Comment:
Should post `null` here to be more explicit (the linter complains about this line).